apollo-federation/tests/query_plan/build_query_plan_tests/mutations.rs

```diff
@@ -0,0 +1,78 @@
+const SUBGRAPH_A: &str = r#"
+      type Foo @key(fields: "id") {
+        id: ID!
+        bar: String
+      }
+
+      type Query {
+        foo: Foo
+      }
+
+      type Mutation {
+        updateFooInA: Foo
+      }
+"#;
+
+const SUBGRAPH_B: &str = r#"
+      type Mutation {
+        updateFooInB: Foo
+      }
+
+      type Foo @key(fields: "id") {
+        id: ID!
+        baz: Int
+      }
+"#;
+
+#[test]
+fn adjacent_mutations_get_merged() {
+    let planner = planner!(
+        SubgraphA: SUBGRAPH_A,
+        SubgraphB: SUBGRAPH_B,
+    );
+    assert_plan!(
+        &planner,
+        r#"
+        mutation TestMutation {
+          updateInAOne: updateFooInA {
+            id
+            bar
+          }
+          updateInBOne: updateFooInB {
+            id
+            baz
+          }
+          updateInATwo: updateFooInA {
+            id
+            baz
+          }
+        }
+        "#,
+        @r###"
+        QueryPlan {
+          Sequence {
+            Fetch(service: "SubgraphA") {
+              {
+                updateInAOne: updateFooInA {
+                  id
+                  bar
+                }
+                updateInATwo: updateFooInA {
+                  id
+                  bar
+                }
+              }
+            },
+            Fetch(service: "SubgraphB") {
+              {
+                updateInBOne: updateFooInB {
+                  id
+                  baz
+                }
+              }
+            },
```

Comment:
This is still incorrect (or at least not what I intend to happen). The QP mistakes each root field as belonging to the same subgraph.